bears/general/KeywordBear.py

```diff
@@ -2,18 +2,67 @@
 
 from coalib.bearlib import deprecate_settings
 from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
 from coalib.results.Result import RESULT_SEVERITY, Result
 
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.results.SourceRange import SourceRange
+
+
+def _get_comments(dependency_results):
+    if not dependency_results:
+        return {}
+
+    annotation_bear_results = dependency_results.get('AnnotationBear')
+    if (not annotation_bear_results or
+            not isinstance(annotation_bear_results, list)):
+        return {}
+
+    for result in annotation_bear_results:
+        yield from result.contents.get('comments', [])
+
+
+def generate_diff(comments, file, filename,
+                  line, line_number, pos):
+    diffs = None
+    todo_source_range = SourceRange.from_values(filename, line_number + 1,
+                                                pos + 1)
+    for comment_sourcerange in comments:
+        if todo_source_range not in comment_sourcerange:
+            continue
+
+        diff = Diff(file)
+        comment_start = comment_sourcerange.start.column
+        line_before_todo_comment = line[:comment_start - 1].strip()
+        # remove just part of the line
+        if (line_before_todo_comment and
+                line_number + 1 == comment_sourcerange.start.line):
+            diff.change_line(
+                line_number + 1,
+                line,
+                line.replace(
+                    line[comment_start - 1:], '').rstrip() + '\n')
+        else:
+            diff.delete_line(line_number + 1)
+
+        diffs = {filename: diff}
```

Comment:
what if `comment_sourcerange` contains more than one item? then you override the last calculated diff. Imo you want to initialize above `diff = {}` (instead of setting to None) and then
```
diffs[filename] = diff
```